File: Contraints/StringIsSetConstraint.vb
' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace Constraints

    Public Class StringIsSetConstraint
        Implements IConstraint(Of String)

        Public Function ValueSatisfiesConstraint(ByVal value As String) As Boolean Implements IConstraint(Of String).ValueSatisfiesConstraint

            Return Not String.IsNullOrEmpty(value)

        End Function

        Public Overrides Function ToString() As String

            Return "string must not be empty"

        End Function

    End Class

End Namespace

File: Contraints/StringIsSetConstraint.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.Constraints
{
	public class StringIsSetConstraint : IConstraint<string>
	{
		public bool ValueSatisfiesConstraint(string value)
		{
			return !string.IsNullOrEmpty(value);
		}
			
		public override string ToString()
		{
			return "string must not be empty";
		}
	}
}